lib/home_widget.dart

```diff
@@ -188,21 +188,27 @@ class HomeWidget {
 
       try {
         late final String? directory;
-        try {
-          // coverage:ignore-start
-          if (Platform.environment.containsKey('FLUTTER_TEST')) {
-            throw UnsupportedError(
-              'Tests should always use default Path provider for easier mocking',
-            );
-          }
+
+        // coverage:ignore-start
+        if (Platform.environment.containsKey('FLUTTER_TEST')) {
+          throw UnsupportedError(
+            'Tests should always use default Path provider for easier mocking',
+          );
+        }
+        if (Platform.isIOS) {
           final PathProviderFoundation provider = PathProviderFoundation();
+          assert(
+            HomeWidget.groupId != null,
+            'No groupId defined. Did you forget to call `HomeWidget.setAppGroupId`',
+          );
           directory = await provider.getContainerPath(
             appGroupIdentifier: HomeWidget.groupId!,
           );
-          // coverage:ignore-end
-        } on UnsupportedError catch (_) {
+        } else {
           directory = (await getApplicationSupportDirectory()).path;
         }
+        // coverage:ignore-end
+
```

Comment:
```suggestion
        // coverage:ignore-start
        if (Platform.isIOS) {
          final PathProviderFoundation provider = PathProviderFoundation();
          assert(
            HomeWidget.groupId != null,
            'No groupId defined. Did you forget to call `HomeWidget.setAppGroupId`',
          );
          directory = await provider.getContainerPath(
            appGroupIdentifier: HomeWidget.groupId!,
          );
        } else {
          // coverage:ignore-end
          directory = (await getApplicationSupportDirectory()).path;
        }
```

Overlooked this during the preview. The check for the `FLUTTER_TEST` makes the test fail as this should always fall back to the `getApplicationSupportDirectory` as that is used for mocking

Reply:
ah, will have to manually remove those lines. Github is not letting me commit suggestion and throws `Applying suggestions on deleted lines is currently not supported. `